src/test/java/org/sonarlint/intellij/config/global/SonarLintGlobalSettingsTest.java

```diff
@@ -47,10 +56,89 @@ public void testRoundTrip() {
 
   @Test
   public void testGetInstance() {
-    SonarLintGlobalSettings instance = new SonarLintGlobalSettings();
-    super.register(app, SonarLintGlobalSettings.class, instance);
-    assertThat(SonarLintGlobalSettings.getInstance()).isEqualTo(instance);
+    SonarLintGlobalSettings settings = new SonarLintGlobalSettings();
+    super.register(app, SonarLintGlobalSettings.class, settings);
+    assertThat(SonarLintGlobalSettings.getInstance()).isEqualTo(settings);
+
+  }
+
+  @Test
+  public void testLoadStateOldFormat() throws Exception {
+    SonarLintGlobalSettings state = new SonarLintGlobalSettings();
+    HashSet<String> includedRules = new HashSet<>();
+    includedRules.add(RULE);
+    HashSet<String> excludedRules = new HashSet<>();
+    excludedRules.add(RULE1);
+    Field includedRulesField = state.getClass().getDeclaredField("includedRules");
+    includedRulesField.setAccessible(true);
+    includedRulesField.set(state, includedRules);
+    Field excludedRulesField = SonarLintGlobalSettings.class.getDeclaredField("excludedRules");
+    excludedRulesField.setAccessible(true);
+    excludedRulesField.set(state, excludedRules);
+    SonarLintGlobalSettings settings = new SonarLintGlobalSettings();
+    Field rulesField = state.getClass().getDeclaredField("rules");
+    rulesField.setAccessible(true);
+
+    settings.loadState(state);
+
+    Map<String, SonarLintGlobalSettings.Rule> rules = (Map<String, SonarLintGlobalSettings.Rule>) rulesField.get(settings);
+    assertThat(rules).containsKey(RULE);
+    assertThat(rules).containsKey(RULE1);
+    assertThat(rules.get(RULE).isActive).isTrue();
+    assertThat(rules.get(RULE1).isActive).isFalse();
+
+  }
+
+  @Test
+  public void testLoadStateNewFormat() throws Exception {
+    SonarLintGlobalSettings state = new SonarLintGlobalSettings();
+    SonarLintGlobalSettings settings = new SonarLintGlobalSettings();
+    HashMap<String, SonarLintGlobalSettings.Rule> rules = new HashMap<>();
+    rules.put(RULE, new SonarLintGlobalSettings.Rule(false));
+    Field rulesField = state.getClass().getDeclaredField("rules");
```

Comment:
I suggest to add a package private accessor, to be used by the test, instead of using reflection.